x-pack/plugins/lens/public/indexpattern_datasource/utils.ts

```diff
@@ -53,9 +54,16 @@ export function isColumnInvalid(
   if (!column) return;
 
   const operationDefinition = column.operationType && operationDefinitionMap[column.operationType];
-  return !!(
-    operationDefinition.getErrorMessage &&
-    operationDefinition.getErrorMessage(layer, columnId, indexPattern)
+  // check also references for errors
+  const referencesHaveErrors = (column as ReferenceBasedIndexPatternColumn).references
```

Comment:
nit: You don't need to cast here, `'references' in column &&` will do it in a type safe way